File: downloadablebookandexercisecode/1256/code_files/Chapter_07_Strings/USACO_2012_nov_b1_FindTheCow.py
total = 0
l_legs = 0
prev_c = ' '
c = ''
f = open('cowfind.in', 'r')
while True:
    c = f.read(1)
    if not c:
        break
    if c == '(' and prev_c == '(':
        l_legs += 1
    if c == ')' and prev_c == ')':
        total += l_legs
    prev_c = c

f_out = open('cowfind.out', 'w')
f_out.write(repr(total))

File: downloadablebookandexercisecode/1256/code_files/Chapter_07_Strings/USACO_2012_nov_b1_FindTheCow.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main() {

    ifstream s_in{"cowfind.in"};

    int total = 0;
    char ch, prev_ch = ' ';
    int l_legs = 0;

    while (s_in >> ch){
        if (ch == '(' && prev_ch == '(') l_legs++;
        if (ch == ')' && prev_ch == ')') total += l_legs;
        prev_ch = ch;
    }


    ofstream s_out{"cowfind.out"};
    s_out << total;
    return 0;
}
